### PR TITLE
Updates RPD Auto-Wrenching

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Auto-wrenching is now a proc.

The user's off-hand tool for automatic RPD wrenching now checks for `TOOL_WRENCH` behaviour instead of type-checking for `obj/item/wrench`.

Cyborgs are now able to auto-wrench with the RPD.

Auto-wrenching now auto-screws transit tubes when a tool with `TOOL_SCREWDRIVER` is held. Previously you'd try to wrench the tubes and destroy them.

## Why It's Good For The Game
Not trying to destroy transit tubes you've just spawned when you have a wrench out is good.

Borgs getting to auto wrench is way better than spam-clicking 500 times on a long pipeline and then having to go back because some sections still didn't wrench.

## Testing
Spawned in a cyborg and a skrell.

Spawned a bunch of different pipes and devices as both - with and without a wrench equipped. Repeated with a screwdriver.

Every category of item spawnable automatically got wrenched when a wrench was held, except for transit tubes.

Transit tubes got automatically screwed down when a screwdriver was held, no other category interacted with it.

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
tweak: Cyborgs can now automatically wrench RPD spawned pipes and devices with a wrench also equipped, just like humans can.
tweak: Auto-wrenching on transit tubes has been replaced with auto-screwing. This means you actually try to secure the tube instead of trying to destroy it. A screwdriver instead of a wrench is required to auto-screw (obviously).
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/rpd.dm [hunk 1]
```diff
@@ -195,6 +191,21 @@
 	QDEL_NULL(P)
 	activate_rpd()
 
+/// Automatically wrenches down an atmos device/pipe if the user has a wrench equipped in their off-hand (or in an inactive module slot as a robot).
+/obj/item/rpd/proc/automatic_wrench_down(mob/living/user, obj/item/target)
```
Comment (Contrabang) [inline code/game/objects/items/weapons/rpd.dm:195]:
```suggestion
/obj/item/rpd/proc/automatic_wrench_down(mob/living/user, obj/item/target, tool_type = TOOL_WRENCH)
```

File: code/game/objects/items/weapons/rpd.dm [hunk 2]
```diff
@@ -195,6 +191,21 @@
...
+		for(var/obj/item/robot_module_item in user.get_all_slots())
+			if(robot_module_item.tool_behaviour == TOOL_WRENCH)
+				off_hand_item = robot_module_item
+				break
+
+	if(!off_hand_item || off_hand_item.tool_behaviour != TOOL_WRENCH)
+		return
+
+	if(user.can_reach(target, off_hand_item))
+		target.wrench_act(user, off_hand_item)
```
Comment (Contrabang) [inline code/game/objects/items/weapons/rpd.dm:207]:
```suggestion
		target.tool_act(user, off_hand_item, tool_type)
```

File: code/game/objects/items/weapons/rpd.dm [hunk 3]
```diff
@@ -195,6 +191,21 @@
 	QDEL_NULL(P)
 	activate_rpd()
 
+/// Automatically wrenches down an atmos device/pipe if the user has a wrench equipped in their off-hand (or in an inactive module slot as a robot).
+/obj/item/rpd/proc/automatic_wrench_down(mob/living/user, obj/item/target)
+	var/obj/item/off_hand_item = user.get_inactive_hand()
+	if(isrobot(user))
+		for(var/obj/item/robot_module_item in user.get_all_slots())
+			if(robot_module_item.tool_behaviour == TOOL_WRENCH)
```
Comment (Contrabang) [inline code/game/objects/items/weapons/rpd.dm:199]:
```suggestion
			if(robot_module_item.tool_behaviour == tool_type)
```

File: code/game/objects/items/weapons/rpd.dm [hunk 4]
```diff
@@ -195,6 +191,21 @@
...
+				off_hand_item = robot_module_item
+				break
+
+	if(!off_hand_item || off_hand_item.tool_behaviour != TOOL_WRENCH)
```
Comment (Contrabang) [inline code/game/objects/items/weapons/rpd.dm:203]:
```suggestion
	if(!off_hand_item || off_hand_item.tool_behaviour != tool_type)
```

File: code/game/objects/items/weapons/rpd.dm [hunk 5]
```diff
@@ -144,8 +142,6 @@
 			S.dir = iconrotation ? iconrotation : user.dir
 
 			to_chat(user, "<span class='notice'>[src] rapidly dispenses [S]!</span>")
```
Comment (Contrabang) [inline code/game/objects/items/weapons/rpd.dm:144]:
```suggestion
			to_chat(user, "<span class='notice'>[src] rapidly dispenses [S]!</span>")
			automatic_wrench_down(user, S, TOOL_SCREWDRIVER)
```

**Review by Contrabang — COMMENTED** [review]
Looks pretty good, having this support the transit pipes with a screwdriver would be pretty cool too

**Comment by Fordoxia** [comment]
> Looks pretty good, having this support the transit pipes with a screwdriver would be pretty cool too

All added and re-tested. Everything is good.

**Review by Contrabang — APPROVED** [review]

**Comment by Fordoxia** [comment]
Closing this because #26855 is an outright superior implementation.

**Pull request closed without merging**